### ISSUE TITLE
Functional Annotation

### ISSUE THREAD

**Issue by kyrenya** [tripal/tripal#689]
Hi, 
I'm trying with no success to follow the tutorial for functional Annotation Modules (https://tripal.info/node/106), though I'm using Tripal v3 (I didnt find a tutorial for this version, is there any?).

I came across several issues: 

_BLAST Module:
 After following the instruction for loading the BLAST results: 
    
        Database: ExPASy SwissProt
        BLAST XML File/Directory: /var/www/html/sites/default/files/Blastx_citrus_sinensis-orange1.1g015632m.g.fasta.0_vs_uniprot_sprot.fasta.out
        Query Type: mRNA
        Parameters: -p blastx -e 1e-6 -m 7
        Submit a job to parse the XML output: checked
        Keywords for custom search: checked
        Click the Save button. You can now see our new Analysis._

I see the Analysis, but when running the job to parse the BLAST results I get the following Error message: 

        PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR: relation "chado.blast_hit_data"  does not exist LINE 1: DELETE FROM chado.blast_hit_data WHERE analysisfeature_id =... ^: DELETE FROM chado.blast_hit_data WHERE analysisfeature_id = :analysisfeature_id ; Array ( [:analysisfeature_id] => 157384 ) in chado_query() (line 1715 of /var/www/html/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).

InterProScan Module: 
Everything seems to work fine, but I don't see the "Interpro Report" link anywhere.
I also cannot follow the link http://[localhost]/orange1.1g015615m

KEGG Module works fine! 

Thanks in advance.

By the way, I'm not using Tripal via git..   should I?

**Comment by bradfordcondon** [comment]
Hello,  

This guide is out of date right now, as these modules will load the Tripal 2 version for each!  The Tripal 3 version of the blast module has renamed that table: that wh y you see that error for tripal 3.  The guide still works because of Tripal legacy mode, but really we need to update this for Tripal 3.

How to proceed:
Upgrade the extension modules to tripal 3.  you can do this by attaching your current modules to git, or deleting them and re-downloading with git.  I provide instructions for both below:

(convert existing modules from drush to tripal 3)
```
cd tripal_analysis_blast
git init
git remote add https://github.com/tripal/tripal_analysis_blast
git pull
git checkout 7.x-3.x
drush updatedb

```

or:

(delete and redownload)
```
rm -rf tripal_analysis_blast
git clone https://github.com/tripal/tripal_analysis_blast
cd tripal_analysis_blast
git checkout 7.x-3.x
drush pm-enable tripal_analysis_blast
```

**Comment by bradfordcondon** [comment]
oops- i see you linked to the tripal.info user's guide.  The tripal 3 guide is here:  https://tripal.readthedocs.io/en/latest/user_guide/example_genomics.html

but youre right i dont think we've written an equivalent guide for using these extension modules in tripal 3 yet.    I think we're waiting for the official Tripal 3 release this week.

Apologies for the confusion!

**Comment by kyrenya** [comment]
Hi Thanks a lot for your fast answer, I'm going to follow your recommendations and I'll keep this post updated with my progress (hopefully there is some progress XD XD). 
Thanks and cheers!

**Comment by kyrenya** [comment]
nop, didnt work for me. 
I started a fresh new start trying to install tripal v2. 
Anyway.. any clue of when is that official Tripal 3 release out? 
Thanks!

**Comment by bradfordcondon** [comment]
hi @kyrenya im sorry it didnt work: you mean still got the `ERROR: relation "chado.blast_hit_data"` error even after updating the blast module to tripal 3? 

We are very close to  the official Tripal 3 release, i would hope this week.

**Comment by kyrenya** [comment]
Hi,  the error disappeared, but I was still unable to go directly to the link  (http://localhost/feature/citrus/sinensis/mRNA/PAC%3A18136219) - (with my localhost), and also no "homology" option in the content anywhere. Interpro module behaved the same, not errors while performing it, but no report anywhere.
KEGG worked fine!

**Comment by bradfordcondon** [comment]
As in you couldnt visit that feature page at all?  Because that would indicate a problem with loading the FASTA file.  If you're saying the BLAST data didnt show up on that feature page, then (for **tripal 3**) you first need to **add the BLAST field** at `admin -> structure -> mRNA -> fields` (click Check for new fields) and then enable the field for display by clicking the display tab.

**Comment by kyrenya** [comment]
aham! ok! I cannot try it now because I'm starting from scratch with Tripal v2...  but once I get to this point again, I will keep you posted. Thanks a lot lot!

**Comment by spficklin** [comment]
Hi @kyrenya.  Just a clarification... the Tripal v3 release should be this week.  However the functional modules are extension modules and those have a different release cycle.  However, we can prioritize getting the remaining fixes and documentation for those.  Those modules are already in use on some Tripal v3 sites so I'm sure whatever bugs you are having can be easily work out. 

If I could make a suggestion.  I think it would be easier to start with Tripal v3 and let us help you work out any issues rather than install Tripal v2 and then upgrade to Tripal v3 later.  It will save you the upgrade hassels.  We already have several sites running in production with the Tripal v3 development branch.  Tripal v3 is a major overhaul of Tripal so naturally there will be some uncaught bugs so if you find those, report them and let us help you work through them we can all make Tripal v3 a bit better.  Something to think on....

**Comment by kyrenya** [comment]
Hi! first of all, thanks for all the support. Ok, I was not sure of how crucial was the change between tripal v2 and tripal v3. As I am a noobie at this, I just thought it was going to be easier for me. I think I will play around with v2 a bit more since I've already installed it and things are going smooth (by now), but anyway I will probably come back to the v3 to build my site and try to figure out what I was doing wrong, with your help. In that sense, do you recommend me to make the fast installation of v3 or installing via git?  Big thanks!

**Comment by spficklin** [comment]
I personally like the manual install because it gives step-by-step instructions and the whole picture of what is going on, but that's my preference.  Either install method should work just fine :-)

**Comment by kyrenya** [comment]
okk! Thanks a lot!!

**Comment by spficklin** [comment]
@kyrenya just FYI.... we are currently working on the functional annotation sections of the User's Guide for Tripal v3. It should be available shortly once the pull request (link above) is merged.

**Comment by kyrenya** [comment]
cool! Yes I was reading right now that announcement! :) great!

**Comment by kyrenya** [comment]
Hi there again! :) 
I started (as you have seen for other issues) with v3 latest, and I would like to add functional annotation. Ive seen that documentation is on-going.. do you recommend me to follow v2 guidelines in the meantime or is somewhere where I can see a pre-documentation guidelines for v3 functional annotation modules? 
Thanks in advance! 
Cheers.

**Comment by bradfordcondon** [comment]
the blast and interpro guides have been merged into master, so i am removing the development branch guide i put up for your @kyrenya please use the guides on master:  https://tripal.readthedocs.io/en/latest/user_guide/example_genomics/func_annots.html?highlight=functional

**Comment by kyrenya** [comment]
Thanks I will!

**Comment by spficklin** [comment]
Since there's been no more conversation on this in a few months I'm going to close this out, but please respond if you'd like it reopened for more discussion.
